calico_containers/calico_ctl/node.py

```diff
@@ -437,14 +443,28 @@ def _setup_ip_forwarding():
 
 
 def node_stop(force):
-    num_endpoints = len(client.get_endpoints(hostname=hostname,
-                                             orchestrator_id=DOCKER_ORCHESTRATOR_ID))
-    if num_endpoints and not force:
-        print "Current host has active endpoints so can't be stopped." \
-              " Force with --force"
-        return
+    """
+    Stop the Calico node.  This stops the containers (calico/node and
+    calico/node-libnetwork) that are started by calicoctl node.
+    """
+    endpoints = len(client.get_endpoints(hostname=hostname,
+                                         orchestrator_id=DOCKER_ORCHESTRATOR_ID))
+    if endpoints:
+        if not force:
+            print_paragraph("Current host has active endpoints so can't be "
+                "stopped.  Force with --force")
+            print_paragraph("Note that stopping the node while there active "
+                            "endpoints may make it difficult to clean up the "
+                            "endpoints: for example, Docker containers "
+                            "networked using libnetwork with Calico will not "
+                            "invoke network cleanup during the normal "
+                            "container lifecycle.")
+            sys.exit(1)
+        else:
+            print_paragraph("Stopping node while host has active endpoints.  "
+                            "If this in error, restart the node using the "
+                            "'calicoctl node' command.")
 
-    client.remove_host(hostname)
     try:
         docker_client.stop("calico-node")
```

Comment:
that's odd.  i thought i'd even included a test for that.  yes it should be there nopt sure why it isn't!